code/modules/loadout/loadout_items.dm

```diff
@@ -0,0 +1,348 @@
+/// Global list of ALL loadout datums instantiated.
+/// Loadout datums are created by loadout categories.
+GLOBAL_LIST_EMPTY(all_loadout_datums)
+
+/// Global list of all loadout categories
+/// Doesn't really NEED to be a global but we need to init this early for preferences,
+/// as the categories instantiate all the loadout datums
+GLOBAL_LIST_INIT(all_loadout_categories, init_loadout_categories())
+
+/// Inits the global list of loadout category singletons
+/// Also inits loadout item singletons
+/proc/init_loadout_categories()
+	var/list/loadout_categories = list()
+	for(var/category_type in subtypesof(/datum/loadout_category))
+		loadout_categories += new category_type()
+
+	sortTim(loadout_categories, /proc/cmp_loadout_categories)
+	return loadout_categories
+
+/proc/cmp_loadout_categories(datum/loadout_category/A, datum/loadout_category/B)
+	var/a_order = A::tab_order
+	var/b_order = B::tab_order
+	if(a_order == b_order)
+		return cmp_text_asc(A, B)
+	return cmp_numeric_asc(a_order, b_order)
+
+/**
+ * # Loadout item datum
+ *
+ * Singleton that holds all the information about each loadout items, and how to equip them.
+ */
+/datum/loadout_item
+	/// The category of the loadout item. Set automatically in New
+	VAR_FINAL/datum/loadout_category/category
```

Comment:
Nit: Feels like this should be at the end so the user provided stuff comes first

Reply:
From past experience people will continue to tack user provided stuff after wherever you put it without a care